src/transformStateWithClones.js

```diff
@@ -7,7 +7,41 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateHistory = [];
+
+  for (const action of actions) {
```

Comment:
Use destruction for `action`
```suggestion
  for (const action of actions) {
    const { type, ...... } = action;
  
```